src/main/java/it/gov/pagopa/payhub/activities/service/exportflow/debtposition/BaseExportFlowFileService.java

```diff
@@ -0,0 +1,180 @@
+import it.gov.pagopa.payhub.activities.dto.export.debtposition.ExportFlowFileResult;
+import it.gov.pagopa.payhub.activities.exception.exportFlow.ExportFlowFileNotFoundException;
+import it.gov.pagopa.payhub.activities.exception.exportFlow.InvalidExportStatusException;
+import it.gov.pagopa.payhub.activities.service.CsvService;
+import it.gov.pagopa.payhub.activities.service.FileArchiverService;
+import it.gov.pagopa.payhub.activities.util.Utilities;
+import it.gov.pagopa.pu.processexecutions.dto.generated.ExportFileStatus;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+@Slf4j
+public abstract class BaseExportFlowFileService<E,F,D,C> {
+
+    private final CsvService csvService;
+    private final Class<C> csvRowDtoClass;
+    private final FileArchiverService fileArchiverService;
+    private final Path workingDirectory;
+    private final String relativeFileFolder;
+    private final String fileNamePrefix;
+
+    protected BaseExportFlowFileService(CsvService csvService, Class<C> csvRowDtoClass, FileArchiverService fileArchiverService, Path workingDirectory, String relativeFileFolder, String fileNamePrefix) {
+        this.csvService = csvService;
+        this.csvRowDtoClass = csvRowDtoClass;
+        this.fileArchiverService = fileArchiverService;
+        this.workingDirectory = workingDirectory;
+        this.relativeFileFolder = relativeFileFolder;
+        this.fileNamePrefix = fileNamePrefix;
+    }
+
+    public ExportFlowFileResult executeExport(Long exportFileId) {
+        int[] pageNumber = {0};
+        long[] exportedRows = {0};
+
+        E exportFileRecord = findExportFileRecord(exportFileId);
+        ExportFileStatus exportStatus = getExportStatus(exportFileRecord);
+
+        if (exportStatus.equals(ExportFileStatus.PROCESSING)){
+            Long organizationId = getOrganizationId(exportFileRecord);
+            String profile = getFlowFileVersion(exportFileRecord);
+            Path csvFilePath = workingDirectory.resolve(String.valueOf(organizationId)).resolve(relativeFileFolder).resolve(getExportFileName(exportFileId));
+            F exportFilter = getExportFilter(exportFileRecord);
+
+            try {
+                csvService.createCsv(csvFilePath, csvRowDtoClass, () ->{
+                            List<C> csvRows = retrieveAndMapPage(exportFileRecord, exportFilter, pageNumber[0]++);
+                            exportedRows[0] += csvRows.size();
+                            return csvRows;
+                        }
+                        , profile);
+            } catch (IOException e) {
+                throw new IllegalStateException("Error writing to CSV file: " + e.getMessage(), e);
+            }
+
+            Path zipFilePath = createZipArchive(csvFilePath);
+
+            return ExportFlowFileResult.builder()
+                    .fileName(zipFilePath.getFileName().toString())
+                    .filePath(zipFilePath.toString())
```

Comment:
fileName should contains just the fileName with the .zip extension (not the .cipher added to the compressAndArchive utility: I don't know if this is you case, but you should be aware of this)

filePath should be the folder path (without the fileName) relative to the organization folder